R/matching.R

```diff
@@ -103,6 +106,10 @@ closest <- function(x, table, tolerance = Inf, ppm = 0,
                     duplicates = c("keep", "closest", "remove"),
                     nomatch = NA_integer_, .check = TRUE) {
 
+    ntolerance <- length(tolerance)
+    if (ntolerance != 1L && ntolerance != length(x))
```

Comment:
I'm wondering - should we put all these tests within a `if (.check)`? I guess there will only be a performance improvement if length of `tolerance` and or `ppm` is > 0.

Reply:
I am not sure about this. I guess it won't decrease the runtime significantly.